### PR TITLE
JAMES-2083 Don't create migration threads when migration not expected

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2Migration.java [hunk 1]
```diff
@@ -64,9 +64,15 @@ public V1ToV2Migration(CassandraMessageDAO messageDAOV1, CassandraMessageDAOV2 m
...
         boolean ensureFifoOrder = false;
         this.messagesToBeMigrated = new ArrayBlockingQueue<>(cassandraConfiguration.getV1ToV2QueueLength(), ensureFifoOrder);
-        IntStream.range(0, cassandraConfiguration.getV1ToV2ThreadCount())
-            .mapToObj(i -> new V1ToV2MigrationThread(messagesToBeMigrated, messageDAOV1, messageDAOV2, attachmentLoader))
-            .forEach(migrationExecutor::execute);
+        executeMigrationThread(messageDAOV1, messageDAOV2, cassandraConfiguration);
+    }
+
+    private void executeMigrationThread(CassandraMessageDAO messageDAOV1, CassandraMessageDAOV2 messageDAOV2, CassandraConfiguration cassandraConfiguration) {
+        if (cassandraConfiguration.isOnTheFlyV1ToV2Migration()) {
```
Comment (quynhn) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2Migration.java]: Is this a temporary solution for now?

Comment (aduprat) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2Migration.java]: No, if we don't want the _ on the fly migration_ in the configuration, then we don't have to create the associated threads.

**Comment by aduprat** [comment]
Build time: 1h43

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
Build time 1h38

**Comment by chibenwa** [comment]
:+1:

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2MigrationThread.java [hunk 2]
```diff
@@ -76,9 +76,7 @@ public void run() {
     }
 
     private Optional<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> poll() {
-        synchronized (messagesToBeMigrated) {
-            return Optional.ofNullable(messagesToBeMigrated.poll());
-        }
+        return Optional.ofNullable(messagesToBeMigrated.poll());
```
Comment (strokyl) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2MigrationThread.java]: Replace poll by take please (poll will ntot block for ever but return after a certain ammount of time) take will block forever until there is data!

**Pull request merged**